osbs/build/spec.py

```diff
@@ -431,3 +431,154 @@ def validate(self):
 
     def __repr__(self):
         return "Spec(%s)" % self.__dict__
+
+
+class BuildUserParams(object):
+    def __init__(self):
+        self.git_uri = BuildParam('git_uri')
+        self.git_ref = BuildParam('git_ref', default=DEFAULT_GIT_REF)
+        self.git_branch = BuildParam('git_branch')
+        self.user = UserParam()
+        self.use_auth = BuildParam("use_auth", allow_none=True)
+        self.platform = BuildParam("platform", allow_none=True)
+        self.component = BuildParam('component')
+        self.name = BuildIDParam()
+        self.build_image = BuildParam('build_image')
+        self.build_imagestream = BuildParam('build_imagestream')
+        self.build_from = BuildParam('build_from')
```

Comment:
I can't work out if these three (build_*) are environment settings or user parameters.

Reply:
I hope you're not expecting me to figure that out, then.

Reply:
Looks fine.